keyboards/dailycraft/sandbox/rev1/keymaps/oled/keymap.c

```diff
@@ -0,0 +1,124 @@
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+
+enum layer_names {
+    _0,
+    _1,
+    _2,
+    _3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_0] = LAYOUT(
+        KC_1, KC_2, KC_3, KC_4, KC_5,
+        KC_6, KC_7, KC_8, KC_9, KC_0
+    ),
+    [_1] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [_2] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [_3] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
+
+#ifdef OLED_ENABLE
+
+void render_layer_state(void) {
+    switch (get_highest_layer(layer_state)) {
+        case _0:
+            oled_write_ln_P(PSTR("Layer: Default"), false);
+            break;
+        case _1:
+            oled_write_ln_P(PSTR("Layer: 1"), false);
+            break;
+        case _2:
+            oled_write_ln_P(PSTR("Layer: 2"), false);
+            break;
+        case _3:
+            oled_write_ln_P(PSTR("Layer: 3"), false);
+            break;
+        default:
+            oled_write_ln_P(PSTR("Layer: Undefined"), false);
+    }
+}
+
+void render_logo(void) {
+    static const char PROGMEM logo[] = {0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94, 0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, 0xb0, 0xb1, 0xb2, 0xb3, 0xb4, 0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, 0xd0, 0xd1, 0xd2, 0xd3, 0xd4, 0};
+    oled_write_P(logo, false);
+}
+
+char keylog_str[24]  = {};
+char keylogs_str[21] = {};
+int  keylogs_str_idx = 0;
+
+const char code_to_name[60] = {' ', ' ', ' ', ' ', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'R', 'E', 'B', 'T', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ';', '\'', ' ', ',', '.', '/', ' ', ' ', ' '};
+
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+    char name = ' ';
+    if (keycode < 60) {
+        name = code_to_name[keycode];
+    }
+
+    // update keylog
+    snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c", record->event.key.row, record->event.key.col, keycode, name);
```

Comment:
https://docs.qmk.fm/#/squeezing_avr?id=oled-tweaks

This will save you ~1.5kb of firmware space.

Reply:
Simplify the OLED implementation.